### PR TITLE
fix _repeat_command()

### PR DESCRIPTION
**Pull request by mhlmhl**
Fix minor bug: the _repeat_command() function is supposed to repeat the specified command N times before treating an error as fatal.  It was not doing that.

### CODE REVIEW

**Review by glossyio — COMMENTED** [review]
@mhlmhl - What is the intent with the _repeat_command() loop?  i.e. 

1. Do we want the ping to fail ONLY ONCE in N (3) tries to report an error and reset the network adapter?  (this was the way it worked previously, right?)
2. Or are we  confirming it needs to fail ALL N (3) times to report an error?

The way this code is working:

1. It runs the `ping` command and checks for NON-ZERO output (which indicates error).
    * If it is ZERO, it exits immediately NO ERROR
    * if it is NON-ZERO, it logs an error message, but continues the FOR loop (with ERROR=false)
2. If the **FINAL for loop only** is NON-ZERO, it will exit and set ERROR=true 

So effectively the ping needs to fail exactly N (3) times in a row.  If it passes even once (even after 2 ping failures) it will return success.

Just wanted to confirm that was what you were intending.  That functionality sounds good to me... 

----

**Edit** - I re-read your comment. It is number 2 above is the intent. It looks good to me. I'll pull it in.

**Review by glossyio — APPROVED** [review]

**Comment by mhlmhl** [comment]
Yes, the comment at the top of the function explains it:
`# try a command N times before declaring that it has failed`

The use case is the PING command.  We found that sometimes a PING would fail initially but then would work when retried. This was causing the wlan_check procedure to reset the network unnecessarily.

So the purpose of this function is to attempt to run the supplied command and return immediately (with zero rc)  if it works. Otherwise, it repeats the command up to the number of times specified, and returns a non-zero result if the command fails every time.

To put it another way, if the PING command succeeds at all, then exit immediately with ok result
